spec/features/admin/admin_authorization_spec.rb

```diff
@@ -0,0 +1,120 @@
+require 'rails_helper'
+
+RSpec.describe 'As an admin' do
+  before(:each) do
+    @admin_user = User.create!(name: "Johnny",
+      street_address: "123 Jonny Way",
+      city: "Johnsonville",
+      state: 'TN',
+      zip_code: 12345,
+      email: "roman@example.com",
+      password: "hamburger01",
+      role: 2
+    )
+
+    @merchant_1 = create(:random_merchant)
+    @merchant_2 = create(:random_merchant)
+    @merchant_3 = create(:random_merchant)
+    @merchant_4 = create(:random_merchant)
+
+    @item_1 = create(:random_item, merchant_id: @merchant_1.id, active?: false)
+    @item_2 = create(:random_item, merchant_id: @merchant_2.id)
+    @item_3 = create(:random_item, merchant_id: @merchant_1.id)
+    @item_4 = create(:random_item, merchant_id: @merchant_2.id)
+
+
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@admin_user)
+  end
+  it "does not allow me to certain pages" do
+
+    expect(@admin_user.role).to eq('admin')
+
+    visit '/merchant'
+    expect(page).to have_content("The page you were looking for doesn't exist (404)")
+
+    visit '/cart'
+    expect(page).to have_content("The page you were looking for doesn't exist (404)")
+  end
+  it "can disable merchants" do
+    visit '/admin/merchants'
+
+    expect(@merchant_1.items.last.active?).to be_truthy
+
+    within "#merchant-#{@merchant_1.id}" do
+      expect(page).to have_button("Disable")
+      click_button("Disable")
+    end
+
+    @merchant_1.items.last.reload
+
+    expect(@merchant_1.items.last.active?).to be_falsey
```

Comment:
here it would have been nice to check that the first `item` for that `merchant` stays `inactive` when disabling the `merchant`